docs/resources/cloud_credentials_v3.md

```diff
@@ -0,0 +1,98 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "rafay_cloud_credentials_v3 Resource - terraform-provider-rafay"
+subcategory: ""
+description: |-
+  
+---
+
+# rafay_cloud_credentials_v3 (Resource)
+
+Cloud credentials version 3
+
+## Example Usage
+
+Credentials Example
+
+```terraform
+resource "rafay_cloud_credentials_v3" "tftestcredentials" {
+  metadata {
+    name    = "tf-credentials-v3"
+    project = "terraform"
+  }
+  spec {
+    type = "ClusterProvisioning"
+    provider = "aws"
+    credentials {
+        type = "AccessKey"
+        access_id = "dummy-id"
+        secret_key = "dummy-key"
+        session_token = "fake-token"
+    } 
+    sharing {
+      enabled = false
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Argument Reference
+
+---
+
+***Required***
+
+- `metadata` - (Block List, Max: 1) Contains data that helps uniquely identify the resource. (See [below for nested schema](#nestedblock--metadata))
+- `spec` - (Block List, Max: 1) Defines the characteristics for the resource. (See [below for nested schema](#nestedblock--spec))
+
+***Optional***
+
+- `timeouts` - (Block) Sets the duration of time the create, delete, and update functions are allowed to run. If the function takes longer than this, it is assumed the function has failed. The default is 10 minutes. (See [below for nested schema](#nestedblock--timeouts))
+
+---
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+***Required***
+
+- `name` - (String) The name of the resource. This must be unique to the organization.
+- `project` - (String) The name of the project.
+
+---
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+***Required***
+
+- `credentials` - (Block List) Contains data for the credentials. (See [below for nested schema](#nestedblock--spec--credentials))
+- `provider` - (String) - The cloud provider. The supported value is: `aws`.
+- `sharing` - (Boolean) - Enables sharing the cloud credentials.
+- `type` - (String) The type of credentials. The supported value is: `ClusterProvisioning`.
+
+<a id="nestedblock--spec--credentials"></a>
+### Nested Schema for `spec.credentials`
+
+***Required***
+
+- `access_id` - (String) The access id for the credentials.
```

Comment:
Not sure why we are restricting to only aws key based credentails. Are we not supporting others? @stephan-rafay ^^